File: Android/java/doraemonkit/src/main/java/com/didichuxing/doraemonkit/kit/network/ui/ListDropDownAdapter.java
package com.didichuxing.doraemonkit.kit.network.ui;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.TextView;

import com.didichuxing.doraemonkit.R;

import java.util.List;



public class ListDropDownAdapter extends BaseAdapter {

    private Context context;
    private List<String> list;
    private int checkItemPosition = 0;

    public List<String> getList() {
        return list;
    }

    public void setCheckItem(int position) {
        checkItemPosition = position;
        notifyDataSetChanged();
    }

    public ListDropDownAdapter(Context context, List<String> list) {
        this.context = context;
        this.list = list;
    }

    @Override
    public int getCount() {
        return list.size();
    }

    @Override
    public Object getItem(int position) {
        return null;
    }

    @Override
    public long getItemId(int position) {
        return position;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        ViewHolder viewHolder;
        if (convertView != null) {
            viewHolder = (ViewHolder) convertView.getTag();
        } else {
            convertView = LayoutInflater.from(context).inflate(R.layout.dk_item_default_drop_down, null);
            viewHolder = new ViewHolder(convertView);
            convertView.setTag(viewHolder);
        }
        fillValue(position, viewHolder);
        return convertView;
    }

    private void fillValue(int position, ViewHolder viewHolder) {
        viewHolder.mText.setText(list.get(position));
        if (checkItemPosition != -1) {
            if (checkItemPosition == position) {
                viewHolder.mText.setTextColor(context.getResources().getColor(R.color.dk_drop_down_selected));
                viewHolder.mText.setBackgroundResource(R.color.dk_check_bg);
            } else {
                viewHolder.mText.setTextColor(context.getResources().getColor(R.color.dk_drop_down_unselected));
                viewHolder.mText.setBackgroundResource(R.color.dk_color_FFFFFF);
            }
        }
    }

    static class ViewHolder {
        TextView mText;

        ViewHolder(View contenView) {
            mText = contenView.findViewById(R.id.text);
        }
    }
}

File: Android/kotlin/doraemonkit/src/main/java/com/didichuxing/doraemonkit/kit/network/ui/ListDropDownAdapter.kt
package com.didichuxing.doraemonkit.kit.network.ui

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.TextView
import com.didichuxing.doraemonkit.R

class ListDropDownAdapter(private val context: Context, val list: List<String>) : BaseAdapter() {
    private var checkItemPosition = 0

    fun setCheckItem(position: Int) {
        checkItemPosition = position
        notifyDataSetChanged()
    }

    override fun getCount(): Int {
        return list.size
    }


    override fun getItemId(position: Int): Long {
        return position.toLong()
    }

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        var convertView = convertView
        val viewHolder: ViewHolder
        if (convertView != null) {
            viewHolder = convertView.tag as ViewHolder
        } else {
            convertView = LayoutInflater.from(context).inflate(R.layout.dk_item_default_drop_down, null)
            viewHolder = ViewHolder(convertView)
            convertView.tag = viewHolder
        }
        fillValue(position, viewHolder)
        return convertView!!
    }

    override fun getItem(p0: Int): Any? {
        return null
    }

    private fun fillValue(position: Int, viewHolder: ViewHolder) {
        viewHolder.mText.text = list[position]
        if (checkItemPosition != -1) {
            if (checkItemPosition == position) {
                viewHolder.mText.setTextColor(context.resources.getColor(R.color.dk_drop_down_selected))
                viewHolder.mText.setBackgroundResource(R.color.dk_check_bg)
            } else {
                viewHolder.mText.setTextColor(context.resources.getColor(R.color.dk_drop_down_unselected))
                viewHolder.mText.setBackgroundResource(R.color.dk_color_FFFFFF)
            }
        }
    }

    internal class ViewHolder(contenView: View) {
        var mText: TextView

        init {
            mText = contenView.findViewById(R.id.text)
        }
    }

}